src/player.cpp

```diff
@@ -3751,6 +3751,20 @@ void Player::addAttacked(const Player* attacked)
 	attackedSet.insert(attacked->guid);
 }
 
+void Player::removeAttacked(const Player* attacked)
+{
+	if (!attacked || attacked == this) {
+		return;
+	}
+
+	auto it = std::find(attackedSet.begin(), attackedSet.end(), attacked->guid);
```

Comment:
A set is implemented as a red-black tree and it's faster to traverse with `attackedSet.find(attacked->guid)` than the generic `std::find`.